src/models/krate.rs

```diff
@@ -539,36 +517,37 @@ impl Crate {
     }
 }
 
-pub struct OwnerInviteEmail {
-    /// The destination email address for this email.
-    recipient_email_address: String,
+/// Details of a newly created invite.
+#[derive(Debug)]
+pub enum NewOwnerInvite {
+    /// The invitee was a [`User`], and they must accept the invite through the
+    /// UI or via the provided invite token.
+    User(User, Secret<String>),
 
-    /// Email body variables.
-    user_name: String,
-    domain: String,
-    crate_name: String,
-    token: SecretString,
+    /// The invitee was a [`Team`], and they were immediately added as an owner.
+    Team(Team),
 }
 
-impl OwnerInviteEmail {
-    pub fn recipient_email_address(&self) -> &str {
-        &self.recipient_email_address
-    }
+/// Error results from a [`Crate::owner_add()`] model call.
+#[derive(Debug, Error)]
+pub enum OwnerAddError {
+    /// An opaque [`BoxedAppError`].
+    #[error("{0}")] // AppError does not impl Error
+    AppError(BoxedAppError),
```

Comment:
A `BoxedAppError` doesn't seem to impl `std::error::Error` - is this intentional?

If it did, this would have been a `#[from]` and `#[error(transparent)]` but I don't think it's a big deal for this usage. It was a bit surprising though.

Reply:
yeah, the error handling still isn't 100% clean. the problem is that there is a wildcard `From` impl between the two, so `BoxedAppError` currently can't implement `Error` itself.